src/components/media-importer.jsx

```diff
@@ -0,0 +1,392 @@
+import React, { useState, useEffect, useRef } from 'react'
+import { useQuery, useQueryClient } from 'react-query'
+import DragAndDrop from './drag-and-drop'
+import { apiDeleteAsset, apiGetAssets, apiRestoreAsset } from '../util/api'
+import './media.scss'
+import { Base64 } from 'js-base64'
+
+const sortString = (field, a, b) => a[field].toLowerCase().localeCompare(b[field].toLowerCase())
+const sortNumber = (field, a, b) => a[field] - b[field]
+
+const REQUESTED_FILE_TYPES = window.location.hash.substring(1).split(',')
+
+// Approx 20 MB
+const FILE_MAX_SIZE = 20000000
+
+// generic media type definitions and substitutions for compatibility
+const MIME_MAP = {
+	// generic types, preferred
+	image: ['image/jpg', 'image/jpeg', 'image/gif', 'image/png'],
+	audio: ['audio/mp3', 'audio/mpeg', 'audio/mpeg3'],
+	video: [], // placeholder
+	model: ['model/obj'],
+
+	// incompatibility prevention, not preferred
+	jpg: ['image/jpg'],
+	jpeg: ['image/jpeg'],
+	gif: ['image/gif'],
+	png: ['image/png'],
+	mp3: ['audio/mp3', 'audio/mpeg', 'audio/mpeg3'],
+	obj: ['application/octet-stream', 'model/obj'],
+}
+
+const SORT_OPTIONS = [
+	{
+		sortMethod: sortString.bind(null, 'name'), // bind the field name to the sort method
+		name: 'Title',
+		field: 'name',
+	},
+	{
+		sortMethod: sortString.bind(null, 'type'), // bind the field name to the sort method
+		name: 'Type',
+		field: 'type',
+	},
+	{
+		sortMethod: sortNumber.bind(null, 'timestamp'), // bind the field name to the sort method
+		name: 'Date',
+		field: 'timestamp',
+	},
+]
+
+const MediaImporter = () => {
+	const queryClient = useQueryClient()
+	const [selectedAsset, setSelectedAsset] = useState(null)
+	const [sortAssets, setSortAssets] = useState(null) // Display assets list
+	const [sortState, setSortState] = useState({
+		sortAsc: false, // Sorted list in asc or desc
+		sortOrder: 0, // List sorting options
+	})
+	const [assetList, setAssetList] = useState({})
+	const [showDeletedAssets, setShowDeletedAssets] = useState(false)
+	const [filterSearch, setFilterSearch] = useState('') // Search bar filter
+
+	const { data: listOfAssets } = useQuery({
+		queryKey: ['media-assets', selectedAsset],
+		queryFn: () => apiGetAssets(),
+		staleTime: Infinity,
+		onSettled: (data) => {
+			if (!data) console.warn('Error in asset retrieval')
```

Comment:
No login errors (401) are currently not being handled when trying to get, post, or delete an asset (could maybe use the Alert component here if data.title == “Invalid Login”?). apiAuthorVerify() in the creator does eventually handle this by showing the user as having been logged out due to inactivity.

Reply:
Added an additional check to see if the `data` payload contains an error, but otherwise deferred to the creator to actually display an error. Replaced the "Logged out due to inactivity" error with a more generic "You are no longer logged in" message to handle multiple use cases.